stages/org.osbuild.cloud-init

```diff
@@ -72,6 +124,10 @@ def main(tree, options):
 
     config_files_dir = f"{tree}/etc/cloud/cloud.cfg.d"
 
+    if "datasource" in config:
+        # add each configured datasource name to a top-level "datasource_list"
+        config["datasource_list"] = list(config["datasource"].keys())
```

Comment:
These are the datasources to be considered, so the order might play a role and then we get into trouble in Go again, no?

Reply:
I wonder whether it wouldn't be better to make the `datasource_list` an explicit part of the schema, rater than creating its content based on defined data sources. Isn't is possible to define `datasource_list` in the configuration without explicitly configuring them (IOW let them use their defaults)? I'm not sure, I'll try to check it as well.

My original intention was to keep the schema in 1:1 mapping to cloud-init configuration (not be smart and not rename configuration options, etc.). I'm not saying that we need stick to it, but it is one thing to consider, as one can use the knowledge of cloud-init config to create appropriate osbuild stage options.

Reply:
My idea was to have an array of config options, but you raise a valid point!

Reply:
Made it a 1:1 mapping now.  You're right, it should be possible to specify a datasource without configuring it, which should fall back to the source's defaults.  Since the datasources aren't arbitrary but predefined and each has its own options, I only added support for the Azure one for now.  We can add more later if we need them.